### PR TITLE
Update remaining checks to support Shadow DOM

### PR DESCRIPTION
**Pull request by marcysutton**
I went through each of the checks to see if there was Shadow DOM coverage, and there were a couple that needed updating:

- header-present
- internal-link-present

There were a few others where I simply added unit tests:
- aria-errormessage
- p-as-heading

And a few I updated to use accessibleTextVirtual:
- focusable-no-name
- implicit
- button-has-visible-text
- has-visible-text

I also did a bunch of test cleanup and fixed up the shadowCheckSetup utility to support a wider range of scenarios.

Closes https://github.com/dequelabs/axe-core/issues/690

### CODE REVIEW

File: lib/checks/navigation/p-as-heading.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-let siblings = Array.from(node.parentNode.children);
+let parent = node.parentNode;
```
Comment (dylanb) [inline lib/checks/navigation/p-as-heading.js]: This change does not seem necessary

File: test/checks/forms/fieldset.js [hunk 2]
```diff
@@ -4,19 +4,11 @@ describe('fieldset', function () {
 	var shadowSupport = axe.testUtils.shadowSupport.v1;
 	var fixtureSetup = axe.testUtils.fixtureSetup;
 
-	var checkContext = {
-		_data: null,
-		data: function (d) {
-			this._data = d;
-		},
-		relatedNodes: function (nodes) {
-			this._relatedNodes = Array.isArray(nodes) ? nodes : [nodes];
```
Comment (dylanb) [inline test/checks/forms/fieldset.js]: Should we make `MockCheckContext` do the same conditional assignment as this is doing?

File: test/checks/keyboard/has-at-least-one-main.js [hunk 3]
```diff
@@ -4,42 +4,53 @@ describe('has-at-least-one-main', function () {
...
 
 	afterEach(function () {
 		fixture.innerHTML = '';
 		checkContext.reset();
+		axe._tree = undefined;
 	});
 
 	it('should return false if no div has role property', function() {
-		var params = checkSetup('<div id = "target">No role</div>');
+		var params = checkSetup('<div id="target">No role</div>');
```
Comment (dylanb) [inline test/checks/keyboard/has-at-least-one-main.js]: this is a `style` change and should not be mixed up with other changes

File: test/checks/keyboard/tabindex.js [hunk 4]
```diff
@@ -13,17 +13,14 @@ describe('tabindex', function () {
 		fixture.appendChild(node);
 
 		assert.isFalse(checks.tabindex.evaluate(node));
-
-
```
Comment (dylanb) [inline test/checks/keyboard/tabindex.js]: style changes

File: test/testutils.js [hunk 5]
```diff
@@ -100,27 +100,34 @@ testUtils.checkSetup = function (content, options, target) {
...
- * @param Node|String 	Stuff to go into the fixture (html or node)
+ * @param Node|String 	Stuff to go into the fixture (html string or DOM Node)
  * @param Node|String 	Stuff to go into the shadow boundary (html or node)
- * @param String  			Target selector for the check, can be inside or outside of Shadow DOM (default: '#target')
  * @param Object  			Options argument for the check (optional, default: {})
+ * @param String  			Target selector for the check, can be inside or outside of Shadow DOM (optional, default: '#target')
  * @return Array
  */
-testUtils.shadowCheckSetup = function (content, shadowContent, targetSelector, options) {
+testUtils.shadowCheckSetup = function (content, shadowContent, options, targetSelector) {
```
Comment (dylanb) [inline test/testutils.js]: Why the switch?

File: lib/checks/navigation/p-as-heading.js (hunk 1 above)
Comment (marcysutton) [reply lib/checks/navigation/p-as-heading.js]: ah, a leftover thing from some experimentation I did. you're right–I thought I restored it, but I guess not.

File: test/checks/forms/fieldset.js (hunk 2 above)
Comment (marcysutton) [reply test/checks/forms/fieldset.js]: It does do it–that's how I got the tests to pass

File: test/testutils.js (hunk 5 above)
Comment (marcysutton) [reply test/testutils.js]: To make it more consistent with `checkSetup`. There are no instances of `targetSelector`, yet there are options passed in. I wanted to preserve the functionality but make the two methods more consistent

**Comment by marcysutton** [comment]
I've updated the PR with a style-specific commit and addressed the other feedback.

**Pull request merged**